sycl/include/CL/sycl/platform.hpp

```diff
@@ -132,6 +137,11 @@ class __SYCL_EXPORT platform {
   shared_ptr_class<detail::platform_impl> impl;
   platform(shared_ptr_class<detail::platform_impl> impl) : impl(impl) {}
 
+  static std::unordered_map<cl_platform_id,
+                            std::weak_ptr<detail::platform_impl>>
+      platform_impls;
+  static std::mutex platform_mutex;
```

Comment:
This isn't going to work properly. Both unordered_map and mutex are non-trivial classes. See [this](https://github.com/intel/llvm/blob/sycl/sycl/doc/GlobalObjectsInRuntime.md) document for more info.